### PR TITLE
[SYSTEMML-1793] Support right indexing on GPU

### PR DESCRIPTION
**Pull request by niketanpansare**
- This PR supports right indexing without requiring sparse to dense conversion of inputs. Note: this PR only supports dense output.
- Also, added RightIndexingTests in gpu package.

@nakul02 can you please review this PR ?

### CODE REVIEW

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1999/

File: src/main/java/org/apache/sysml/runtime/matrix/data/LibMatrixCUDA.java [hunk 1]
```diff
@@ -3086,14 +3086,67 @@ private static int toInt(long num) throws DMLRuntimeException {
 	//**************** Matrix Manipulation Functions *********************/
 	//********************************************************************/
```
Comment (nakul02) [inline src/main/java/org/apache/sysml/runtime/matrix/data/LibMatrixCUDA.java]: A little javadoc here would be nice.

File: src/main/cpp/kernels/SystemML.cu [hunk 2]
```diff
@@ -28,6 +28,27 @@ nvcc -ptx -arch=sm_30 SystemML.cu
 
 
 /**
+ * Performs a slice operation where the input matrix is sparse and the output matrix is dense.
+ * This function avoids unnecessary sparse to dense conversion of the input matrix.
+ */
+extern "C"
```
Comment (nakul02) [inline src/main/cpp/kernels/SystemML.cu]: A little javadoc explaining the overall logic and the parameters would be nice here.

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/2004/

**Pull request closed without merging**